### PR TITLE
Move over legacy makefile-based unit tests to CMake

### PR DESCRIPTION
**Pull request by avtolstoy**
### Description

As title says.

### Steps to Test

CI build should succeed.

### Example App

N/A

### References

N/A

---

### Completeness

- [x] User is totes amazing for contributing!
- [ ] Contributor has signed CLA ([Info here](https://github.com/spark/firmware/blob/develop/CONTRIBUTING.md))
- [ ] Problem and Solution clearly stated
- [ ] Run unit/integration/application tests on device
- [ ] Added documentation
- [ ] Added to CHANGELOG.md after merging (add links to docs and issues)

### CODE REVIEW

**Comment by avtolstoy** [comment]
Just noticed that I forgot to add `eeprom_emulation.cpp`, I'll fix it.

**Review by tstellanova — APPROVED** [review]

File: test/unit_tests/services/eeprom_emulation.cpp [hunk 1]
```diff
@@ -1184,7 +1185,7 @@ TEST_CASE("Migration from legacy format", "[eeprom]")
     TestEEPROM eeprom;
 
     // Load EEPROM extracted from a Photon into memory
-    loadEEPROMFromFile("eeprom_page1.bin", eeprom, PageBase1, PageSize1);
+    loadEEPROMFromFile(TEST_DIRECTORY "/eeprom_page1.bin", eeprom, PageBase1, PageSize1);
```
Comment (sergeuz) [inline test/unit_tests/services/eeprom_emulation.cpp:1188]: I suggest we move these to a subdirectory in the test suite's source directory, e.g. `fixtures`.

**Review by sergeuz — APPROVED** [review]
Awesome, thank you!

File: test/unit_tests/services/eeprom_emulation.cpp (hunk 1 above)
Comment (avtolstoy) [reply test/unit_tests/services/eeprom_emulation.cpp:1188]: :+1: Agreed. Let's do that.

**Pull request merged**